Consistent encoder names

russcam opened the pull request:
This commit updates encoder names to be consistent within documentation
and align with snake casing convention.

cbuescher commented on docs/plugins/analysis-phonetic.asciidoc:
```diff
@@ -17,9 +17,9 @@ The `phonetic` token filter takes the following settings:
+    `double_metaphone`, `soundex`, `refined_soundex`, `caverphone1`,
     `caverphone2`, `cologne`, `nysiis`, `koelnerphonetik`, `haasephonetik`,
-    `beidermorse`, `daitch_mokotoff`.
+    `beider_morse`, `daitch_mokotoff`.
```
If this is changes, `beidermorde` in L28 should be changed as well. Also there is an occurence in plugins/analysis-phonetic/src/test/resources/org/elasticsearch/index/analysis/phonetic-1.yml and plugins/analysis-phonetic/src/test/java/org/elasticsearch/index/analysis/AnalysisPhoneticFactoryTests.java where we should then also test with this name, although the other variant is supported as well.

cbuescher commented on docs/plugins/analysis-phonetic.asciidoc:
```diff
@@ -17,9 +17,9 @@ The `phonetic` token filter takes the following settings:
 
     Which phonetic encoder to use.  Accepts `metaphone` (default),
-    `doublemetaphone`, `soundex`, `refinedsoundex`, `caverphone1`,
+    `double_metaphone`, `soundex`, `refined_soundex`, `caverphone1`,
```
Maybe also change "refinedsoundec" in plugins/analysis-phonetic/src/test/resources/org/elasticsearch/index/analysis/phonetic-1.yml then

cbuescher replied:
Mabe also change `doublemetaphone` in plugins/analysis-phonetic/src/test/java/org/elasticsearch/index/analysis/AnalysisPhoneticFactoryTests.java and plugins/analysis-phonetic/src/test/resources/org/elasticsearch/index/analysis/phonetic-1.yml then

elasticmachine commented:
Pinging @elastic/es-search-aggs

cbuescher commented:
@russcam just checking in, are you still interested in making this change? If not, I'd like to close this PR.

russcam commented:
I'd still like to make this change @cbuescher. As far as I have seen whilst working on the .NET client, the preference seems to be towards snake-casing, but perhaps @clintongormley has the authoritative answer?

Irrespective of what the _correct_ name should be, I'd still like to make the change, to make the documentation consistent with itself e.g. if `beider_morse` is used in one place in docs, it should be used in all in docs.

clintongormley commented:
My personal preference would be snakecase

cbuescher commented:
@russcam any chance to get this one ready to merge?

russcam commented:
thanks for the ping @cbuescher, I'll take a look first thing tomorrow 👍

russcam commented:
I've addressed the comments in relation to updating names in phonetic-1.yml and AnalysisPhoneticFactoryTests.java to use snake_casing. @cbuescher would you mind taking another look when you get a chance?

cbuescher commented on plugins/analysis-phonetic/src/test/java/org/elasticsearch/index/analysis/AnalysisPhoneticFactoryTests.java:
```diff
@@ -33,8 +33,8 @@ public AnalysisPhoneticFactoryTests() {
-        filters.put("beidermorse", PhoneticTokenFilterFactory.class);
-        filters.put("doublemetaphone", PhoneticTokenFilterFactory.class);
+        filters.put("beider_morse", PhoneticTokenFilterFactory.class);
+        filters.put("double_metaphone", PhoneticTokenFilterFactory.class);
```
Sorry, I was wrong with my previous comment that this needs changing. This will actually fail the test now. I think the test is based on class name lookup, and the "_" doesn't work with this.

russcam commented:
Thanks @cbuescher. Are you OK for me to merge and backport (back to 6.3)?

russcam commented:
ported to 6.x: 990afd0bd80cd33e70596d17de73e5661ce4dcad
